apps/src/code-studio/pd/international_optin/InternationalOptin.jsx

```diff
@@ -0,0 +1,248 @@
+            name: 'lastName',
+            label: labels.lastName,
+            type: 'text',
+            required: true
+          })
+        }
+        {
+          this.buildFieldGroup({
+            name: 'email',
+            label: labels.email,
+            type: 'text',
+            value: this.props.accountEmail,
+            readOnly: true
+          })
+        }
+        {
+          this.buildFieldGroup({
+            name: 'emailAlternate',
+            label: labels.emailAlternate,
+            type: 'text'
+          })
+        }
+        {
+          this.buildButtonsFromOptions({
+            name: 'gender',
+            label: labels.gender,
+            type: 'radio',
+            required: true
+          })
+        }
+        {
+          this.buildFieldGroup({
+            name: 'schoolName',
+            label: labels.schoolName,
+            type: 'text',
+            required: true
+          })
+        }
+        {
+          this.buildFieldGroup({
+            name: 'schoolCity',
+            label: labels.schoolCity,
+            type: 'text',
+            required: true
+          })
+        }
+        {
+          this.buildButtonsFromOptions({
+            name: 'schoolCountry',
+            label: labels.schoolCountry,
+            type: 'radio',
+            required: true
+          })
+        }
+        {
+          this.buildButtonsFromOptions({
+            name: 'ages',
+            label: labels.ages,
+            type: 'check',
+            required: true
+          })
+        }
+        {
+          this.buildButtonsFromOptions({
+            name: 'subjects',
+            label: labels.subjects,
+            type: 'check',
+            required: true,
+            includeOther: true
+          })
+        }
+        {
+          this.buildButtonsFromOptions({
+            name: 'resources',
+            label: labels.resources,
+            type: 'check',
+            required: false,
+            includeOther: true
+          })
+        }
+        {
+          this.buildButtonsFromOptions({
+            name: 'robotics',
+            label: labels.robotics,
+            type: 'check',
+            required: false,
+            includeOther: true
+          })
+        }
+        <FormGroup
+          id="date"
+          controlId="date"
+          validationState={this.getValidationState("date")}
+        >
+          <Row>
+            <Col md={6}>
+              <ControlLabel>
+                Date
+                <span style={{color: 'red'}}> *</span>
+              </ControlLabel>
+            </Col>
+          </Row>
+          <Row>
+            <Col md={6}>
+              <DatePicker
+                date={date}
+                minDate={moment()}
+                onChange={this.handleDateChange}
+                readOnly={false}
+              />
+            </Col>
+          </Row>
+        </FormGroup>
+        {
+          this.buildSelectFieldGroupFromOptions({
```

Comment:
Even though these dropdowns are shown with a default selection, pressing Submit causes them to be highlighted as required, indicating that there's no known selection at this point.  Is there a preferred solution?

Reply:
There is no default value in React component props, and it will render as [empty string](https://github.com/code-dot-org/code-dot-org/blob/b09fe594b804cb674c24237df812279b15415c66/apps/src/code-studio/pd/form_components/FormComponent.jsx#L143) initially. I think at that point, the browser shows the first element selected but it doesn't trigger a React event so doesn't set the value. Either initialize `formData` in the controller to have these initial values, or add a [placeholder](https://github.com/code-dot-org/code-dot-org/blob/b09fe594b804cb674c24237df812279b15415c66/apps/src/code-studio/pd/form_components/FormComponent.jsx#L97-L98) here